src/bodypart.cpp

```diff
@@ -80,6 +80,80 @@ generic_factory<body_part_type> body_part_factory( "body part" );
 
 } // namespace
 
+bool is_legacy_bodypart_id( const std::string &id )
+{
+    static const std::vector<std::string> legacy_body_parts = {
+        "TORSO",
+        "HEAD",
+        "EYES",
+        "MOUTH",
+        "ARM_L",
+        "ARM_R",
+        "HAND_L",
+        "HAND_R",
+        "LEG_L",
+        "LEG_R",
+        "FOOT_L",
+        "FOOT_R",
+        "NUM_BP",
+    };
+
+    for( const auto &string : legacy_body_parts ) {
+        if( string == id ) {
+            return true;
+        }
+    }
+    return false;
+}
+
+const std::string &bodypart_str_id_to_readable( const bodypart_str_id &id )
+{
+    static const std::unordered_map<bodypart_str_id, std::string> bodyparts = {
+        { bodypart_str_id( "torso" ), "Torso" },
+        { bodypart_str_id( "head" ), "Head" },
+        { bodypart_str_id( "eyes" ), "Eyes" },
+        { bodypart_str_id( "mouth" ), "Mouth" },
+        { bodypart_str_id( "arm_l" ), "L. Arm" },
+        { bodypart_str_id( "arm_r" ), "R. Arm" },
+        { bodypart_str_id( "hand_l" ), "L. Hand" },
+        { bodypart_str_id( "hand_r" ), "R. Hand" },
+        { bodypart_str_id( "leg_l" ), "L. Leg" },
+        { bodypart_str_id( "leg_r" ), "R. Leg" },
+        { bodypart_str_id( "foot_l" ), "L. Foot" },
+        { bodypart_str_id( "foot_r" ), "R. Foot" },
+        { bodypart_str_id( "num_bp" ), "All" },
+    };
+
+    if( bodyparts.find( id ) != bodyparts.end() ) {
```

Comment:
Why don't you store the result of `find` and return it directly when the entry is found. Instead of looking it up (via `at`) again?